hls4ml/writer/vivado_writer.py

```diff
@@ -171,10 +179,18 @@ def write_project_cpp(self, model):
                     newline += indent + '#pragma HLS INTERFACE ap_vld port={},{} \n'.format(
                         ','.join(all_inputs), ','.join(all_outputs)
                     )
-                    if model.config.pipeline_style.lower() == 'dataflow':
-                        newline += indent + '#pragma HLS DATAFLOW \n'
+
+                    model_cfg = model.config.get_config_value('HLSConfig')['Model']
+                    if (
```

Comment:
It seems that if unrolled resource conv layers are used, dataflow will not be used. Also, if `n_partition` is not 1, the global II won't the be larger than the internal dense II (`reuse_factor`)?